test/brotli.spec.ts

```diff
@@ -1,234 +1,382 @@
+// polyfill TextEncoder for node < v12
+import { TextEncoder, TextDecoder } from 'text-encoding';
+
+if (typeof global.TextEncoder === 'undefined') {
+    global.TextEncoder = TextEncoder;
+    global.TextDecoder = TextDecoder;
+}
+
 import { expect } from 'chai';
 import * as brotliPromise from '..';
+import init from '../pkg.web/brotli_wasm';
+import * as brotliWeb from '../pkg.web/brotli_wasm';
+import * as fs from 'fs';
+import * as path from 'path';
+
+const webWasmPath = path.join(__dirname, '../pkg.web/brotli_wasm_bg.wasm');
+
+describe('Brotli-wasm', () => {
+    let brotli!: typeof import('..');
+    for (let i = 0; i < 2; i++) {
+        let bundlerToUse = i === 0 ? 'bundler' : 'web (vite compatible)';
+        let bundlerTestName = `Brotli ${bundlerToUse}`;
+        describe(bundlerTestName, function () {
+            beforeEach(async () => {
+                if (i === 0) {
+                    brotli = await brotliPromise;
+                } else {
+                    brotli = brotliWeb;
+                }
+                if (typeof fs.readFileSync === 'undefined') {
+                    await init();
+                } else {
+                    await init(fs.readFileSync(webWasmPath));
```

Comment:
So this means we're now calling `init()` from `pkg.web` during testing for node & webpack too?

That doesn't seem right - we shouldn't test these by running setup code differently to how it's intended/documented to be used, that'll create problems later. It should be just `await brotliPromise` then synchronous `brotli.X()` calls.

It's an easy fix though - we just need to move this inside the previous `else` condition to avoid that. Alternatively this too might be solved with the conditional exports + wrapper changes, which would let us skip the `init()` call entirely for all targets.

Reply:
Yes, will fix, agreed